gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -386,16 +400,18 @@ private String startResumableUpload() throws InterruptedException, IOException {
 
       SimpleResponseObserver<StartResumableWriteResponse> responseObserver =
           new SimpleResponseObserver<>();
-
-      stub.withDeadlineAfter(START_RESUMABLE_WRITE_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS)
-          .startResumableWrite(request, responseObserver);
-      responseObserver.done.await();
-
-      if (responseObserver.hasError()) {
-        throw new IOException(
-            String.format("Failed to start resumable upload for '%s'", resourceId),
-            responseObserver.getError());
-      }
+      runWithRetries(
```

Comment:
Can you use [ResilientOperation](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/util/src/main/java/com/google/cloud/hadoop/util/ResilientOperation.java) instead?

Reply:
Done. It doesn't reduce amount of codes but I guess it's good to have nice features like backoff etc.,